### DISCUSSION TITLE
Hitsounds for beatmaps as packs and separated from the .osu

### DISCUSSION THREAD

**Discussion by HyperLan-git** [ppy/osu#14828] [Ideas]
Proposal :
Have the hitsounds of a map be separated from the hitobjects data. Make a way to swap between the hitsound files (could be .osh) and a way to swap the hitsound samples from the game without modifying the map.
Concretely as a mapper I want to be able to make multiple hitsound packs with, for example, a pack for people who only use default hitsounds, a pack with custom hitsounds and a pack for people who only want to use a specific skin's hitsounds. We could also imagine a scenario where a map has really bad hitsounds and someone could make a hitsound pack to make the map enjoyable for people who play with those.

I know it would take a lot of effort namely :
- rework the beatmap file format and hitsound samples naming
- make an ingame selector
- modify the editor to be able to export such packs
- probably modify the site as well to account for the fact that someone who did not participate in the mapping can make a pack

However it might be worth it in the far future because it is a QOL improvement for people who already specialize in making hitsounds for entire mapsets, it will fix the need for external tools to import hitsounds (the mapping tools thingy) and overall give more importance to what feels to me like the most musical aspect of the mapping process.

**Comment by peppy** [comment]
separating from hitobjects is not a good idea at all and quite the opposite direction to what i'm aiming for. but there will be a method to copy between difficulties.

hitsound "packs" can probably use the skin editor that will come later.
